Makefile

```diff
@@ -42,19 +42,29 @@ stylelint:
 unit-test:
 	npm run test:cov
 
-end-to-end-test:
+shared:
 	sudo docker build -f Dockerfile.nodejs --cache-from welder/web-nodejs:latest -t welder/web-nodejs:latest .
-
-	sudo docker network create welder
-	sudo docker build --cache-from welder/web:latest -t welder/web:latest .
-	sudo docker build -f Dockerfile.with-coverage --cache-from welder/web-with-coverage:latest -t welder/web-with-coverage:latest .
 	sudo docker build -f ./test/end-to-end/Dockerfile --cache-from welder/web-e2e-tests:latest -t welder/web-e2e-tests:latest ./test/end-to-end/
-
+	sudo docker network create welder
 	wget --progress=dot:giga https://s3.amazonaws.com/weldr/metadata.db
 	sudo docker run -d --name api --restart=always -p 4000:4000 -v bdcs-recipes-volume:/bdcs-recipes -v `pwd`:/mddb --network welder --security-opt label=disable welder/bdcs-api-rs:latest
+
+end-to-end-test: shared
+	sudo docker build --cache-from welder/web:latest -t welder/web:latest .
+	sudo docker build -f Dockerfile.with-coverage --cache-from welder/web-with-coverage:latest -t welder/web-with-coverage:latest .
```

Comment:
Remove `--cache-from welder/web-with-coverage:latest` b/c this base image isn't downloaded in CI. This has been a mistake in the previous implementation (copy & paste programming).